lib/matplotlib/tests/test_dates.py

```diff
@@ -1226,3 +1232,58 @@ def test_julian2num():
     mdates.set_epoch('1970-01-01T00:00:00')
     assert mdates.julian2num(2440588.5) == 1.0
     assert mdates.num2julian(2.0) == 2440589.5
+
+
+def test_rcparams_timezone():
+    assert mdates._get_rc_timezone() == mdates.UTC
+    tz = 'Europe/Iceland'
+    assert mdates._get_rc_timezone() != dateutil.tz.gettz(tz)
+
+    plt.rcParams['timezone'] = tz
+
+    assert mdates._get_rc_timezone() == dateutil.tz.gettz(tz)
+
+
+def test_datestr2num():
+    assert mdates.datestr2num('2022-01-10') == 19002.0
+    dt = datetime.date(year=2022, month=1, day=10)
+    assert mdates.datestr2num('2022-01', default=dt) == 19002.0
+    assert np.all(mdates.datestr2num(
+        ['2022-01', '2022-02'], default=dt
+        ) == np.array([19002., 19033.]))
+    assert mdates.datestr2num([]).size == 0
+    assert mdates.datestr2num([], datetime.date(year=2022,
+                                                month=1, day=10)).size == 0
+
+
+def test_concise_formatter_exceptions():
+    locator = mdates.AutoDateLocator()
+    with pytest.raises(ValueError):
```

Comment:
Could you use the `matches='...'` argument in each of these so it's clear what the error being caught is in each case.

Reply:
OK! I'll have a look at that.